core/file_unix.cpp

```diff
@@ -154,11 +151,22 @@ bool File::canonical(const std::string& path, std::string* resolved) {
   free(result);
   return true;
 }
+
 long File::freespace_for_path(const string& path) {
+#if defined(__sun)
```

Comment:
I also changed it to check the fragment size rather than the blocksize, this is because ZFS has variable blocksize and f_frsize is the minimum blocksize and f_bsize is the maximum blocksize. Using f_bsize of ZFS caused the function to overflow, as the maximum blocksize was massive.